devel/lazarus/Portfile

```diff
@@ -4,167 +4,256 @@ PortSystem          1.0
+                set defaultCompiler unsupported
+            }
+        }
+
+        reinplace "s|/usr/local/bin/_PPCARCH_|${prefix}/bin/${defaultCompiler}|g" ${patchtarget}
+        reinplace "s|/usr/local/share|${prefix}/share|g"              ${patchtarget}
+        reinplace "s|/Developer/lazarus|${prefix}/share/lazarus|g"    ${patchtarget}
+
+        reinplace "s|<DebuggerFilename Value=\"/usr/bin/lldb\"/>|<DebuggerFilename Value=\"/usr/bin/lldb\">\\
+            <History Count=\"3\">\\
+              <Item1 Value=\"/usr/bin/lldb\"/>\\
+              <Item2 Value=\"/usr/bin/gdb\"/>\\
+              <Item3 Value=\"${prefix}/bin/gdb\"/>\\
+            </History>\\
+          </DebuggerFilename>|g" ${patchtarget}
+
+    # some more fixes for nogui taking cocoa as a template
+    # if cocoa or nogui get updated, the files in ${filespath} need to be updated manually
+
+        copy ${filespath}/nogui ${worksrcpath}/components/lclextensions/include/
+
+        copy ${worksrcpath}/components/virtualtreeview/units/cocoa         ${worksrcpath}/components/virtualtreeview/units/nogui
+        copy ${worksrcpath}/components/virtualtreeview/include/intf/cocoa/ ${worksrcpath}/components/virtualtreeview/include/intf/nogui
+
+        if {[variant_isset gtk2]} {
+            reinplace "s|/sw/lib;/sw/lib/pango-ft219|${prefix}|g" ${worksrcpath}/lcl/interfaces/lcl.lpk
+            reinplace "s|/sw/lib|${prefix}/lib|g"                 ${worksrcpath}/lcl/interfaces/lcl.lpk
+        }
+
+    # macOS 10.13 (High Sierra) and lower has no separate framework UserNotifications
+        if {${os.major} <= 17} {
+            reinplace "s| -k-weak_framework -kUserNotifications||g" ide/Makefile
+            reinplace "s| -k-weak_framework -kUserNotifications||g" ide/Makefile.fpc
+            reinplace "s| -k-weak_framework -kUserNotifications||g" components/chmhelp/lhelp/Makefile
+            reinplace "s| -k-weak_framework -kUserNotifications||g" components/chmhelp/lhelp/Makefile.fpc
+            reinplace "s| -weak_framework UserNotifications||g" lcl/interfaces/lcl.lpk
+        }
 
-# macOS 10.13 (High Sierra) and lower has no separate framework UserNotifications
-    if {${os.major} <= 17} {
-        reinplace "s| -k-weak_framework -kUserNotifications||g" ide/Makefile
-        reinplace "s| -k-weak_framework -kUserNotifications||g" ide/Makefile.fpc
-        reinplace "s| -k-weak_framework -kUserNotifications||g" components/chmhelp/lhelp/Makefile
-        reinplace "s| -k-weak_framework -kUserNotifications||g" components/chmhelp/lhelp/Makefile.fpc
-        reinplace "s| -weak_framework UserNotifications||g" lcl/interfaces/lcl.lpk
+    # ld segfaults on macOS 15.4 and later, fall back to classic linker
+    # See https://gitlab.com/freepascal.org/lazarus/lazarus/-/issues/41570
+        if {${os.major} >= 24} {
+            reinplace "s|_OPT+=-k-|_OPT+=-k-ld_classic -k-|" \
+                ide/Makefile ide/Makefile.fpc components/chmhelp/lhelp/Makefile
+        }
```

Comment:
one other question: are you sure this is still needed? I removed it a while ago for the `ffmpeg*` ports (see 277ccd60c30c9e836c7b1e9054f2ea59473bd895) because it was causing different issues downstream. And also, it didn't appear to be needed anymore in newer OSes / XCode versions.